README.md

```diff
@@ -47,13 +47,27 @@ You can also set `-Zmiri-start-fn` to make Miri start evaluation with the
 
 ## Running Miri on your own project('s test suite)
 
-Install Miri as a cargo subcommand with `cargo install --all-features`, and install
-a full libstd as described above.
+Install Miri as a cargo subcommand with `cargo install --all-features --path .`.
 
-Then, inside your own project, use `MIRI_SYSROOT=~/.xargo/HOST cargo +nightly
-miri` to run your project, if it is a bin project, or run
-`MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri test` to run all tests in your
-project through Miri.
+Compile your project and its dependencies against a MIR-enabled libstd as described
+above. The easiest way to do this is to copy `xargo/build.sh` and `xargo/Xargo.toml`
+into your own project's root, and to re-run `build.sh`. This ensures that `libstd`
```

Comment:
I am a bit surprised by this. `build.sh` installs the new libstd globally, so you can use `MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri` even without copying it. If you are using the same compiler, this locally built libstd should be compatible with the one shipped with rustc.

Reply:
Ah! If you run `cargo clean` first you are correct. But `cargo` will use cached dependency compilations even if you change `MIRI_SYSROOT`. (Maybe this is a bug in cargo?)

Reply:
I don't see how your instructions can make any difference there. If you used `xargo build` later, that would take the new libstd into account, but `cargo build` for sure doesn't care about `~/.xargo/HOST`.

Something doesn't add up here.

Reply:
I think there are actually some bugs here, and I opened https://github.com/solson/miri/issues/486 to track that.

But none of that should require a copy of `Xargo.toml`, and I have yet to see a situation where that *helps*.

Reply:
In my project (rust-bitcoin), if I run
```
cargo clean
MIRI_SYSROOT=~/.xargo/HOST cargo +nightly-2018-10-14 build
MIRI_SYSROOT=~/.xargo/HOST cargo +nightly-2018-10-14 miri  ## does not error, seems to do nothing
MIRI_SYSROOT=~/.xargo/HOST cargo +nightly-2018-10-14 miri test
```
I will reliably get the error
```
error[E0460]: found possibly newer version of crate `std` which `bitcoin_bech32` depends on                                                                                                                                                                                               
```

My fix was to bring `Xargo.toml` into the crate root and run the commands in `build.sh` there. But now I see that a more straightforward thing to do is to just run `cargo clean` before `MIRI_SYSROOT=~/.xargo/HOST cargo miri test` and it'll work.

This definitely seems like some sort of bug rather than something we should update the README for.